src/Symfony/Component/Encryption/Sodium/SodiumKey.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Encryption\Sodium;
+
+use Symfony\Component\Encryption\Exception\InvalidKeyException;
+use Symfony\Component\Encryption\KeyInterface;
+
+final class SodiumKey implements KeyInterface
```

Comment:
shouldnt those key implementations be internal?

Reply:
Yes, lets!